dali/kernels/slice/slice_flip_normalize_permute_pad_gpu.h

```diff
@@ -101,12 +102,26 @@ class SliceFlipNormalizePermutePadGpu {
     se.add<mm::memory_kind::pinned, detail::SampleDesc<Dims>>(num_samples);
     se.add<mm::memory_kind::device, detail::SampleDesc<Dims>>(num_samples);
 
+    int blocks_per_sm_;
+    CUDA_CALL(cudaOccupancyMaxActiveBlocksPerMultiprocessor(&blocks_per_sm_,
+      detail::SliceFlipNormalizePermutePadKernel<false, false, false, OutputType, InputType, Dims>,
+      kBlockDim,
+      0));
+
     block_count_ = 0;
+    auto number_of_blocks = GetSmCount() * blocks_per_sm_;
+    size_t all_sample_sizes = 0;
     for (auto &elem : args) {
       size_t sample_size = volume(elem.shape);
+      all_sample_sizes += sample_size;
       block_count_ += std::ceil(
         sample_size / static_cast<float>(kBlockSize));
     }
+    if (block_count_ % number_of_blocks != 0) {
+      block_count_ += (number_of_blocks - (block_count_ % number_of_blocks));
+    }
+    kBlockSize = div_ceil(all_sample_sizes, block_count_);
+    block_count_ = std::ceil(all_sample_sizes / static_cast<float>(kBlockSize));
```

Comment:
Also the tests seem not to catch it, but we'll have division by zero here if all samples have size 0. Not sure if it's a real problem though.